### PR TITLE
make vX.X.X-gke.X understand the right gs bucket

### PR DESCRIPTION
**Pull request by krzyzacy**
/assign @fejta @ixdy

### CODE REVIEW

File: kubetest/extract.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (l *extractStrategies) Set(value string) error {
 		`^ci/(.+)$`:                 ci,
 		`^release/(latest.*)$`:      rc,
 		`^release/(stable.*)$`:      stable,
-		`^(v\d+\.\d+\.\d+[\w.-]*)$`: version,
+		`^(v\d+\.\d+\.\d+[\w\.\-+]*)$`: version,
```
Comment (krzyzacy) [inline kubetest/extract.go]: (I guess no-body was really using this code path... added some more tests)

**Comment by krzyzacy** [comment]
ref https://github.com/kubernetes/test-infra/pull/4889, this might be useful

**Comment by krzyzacy** [comment]
PTAL, can I get this in so we can use `v1.8.0-gke.1` rather than `gs://kubernetes-release-gke/release/v1.8.0-gke.1` in extract?

File: kubetest/extract.go [hunk 2]
```diff
@@ -386,7 +386,10 @@ func (e extractStrategy) Extract(project, zone string) error {
 	case version:
 		var url string
 		release := e.option
-		if strings.Contains(release, "+") {
+		re := regexp.MustCompile(`(v\d+\.\d+\.\d+-gke.\d+)$`) // v1.8.0-gke.0
```
Comment (ixdy) [inline kubetest/extract.go]: this special case is a worrying precedent, though I'm not sure how to address it.

File: kubetest/extract.go [hunk 3]
```diff
@@ -68,16 +68,16 @@ func (l *extractStrategies) String() string {
...
-		`^(gs://.*)$`:               gcs,
-		`^(bazel/.*)$`:              bazel,
+		`^(local)`:                     local,
+		`^gke-?(staging|test)?$`:       gke,
+		`^gci/([\w-]+)$`:               gci,
+		`^gci/([\w-]+)/(.+)$`:          gciCi,
+		`^ci/(.+)$`:                    ci,
+		`^release/(latest.*)$`:         rc,
+		`^release/(stable.*)$`:         stable,
+		`^(v\d+\.\d+\.\d+[\w\.\-+]*)$`: version,
```
Comment (ixdy) [inline kubetest/extract.go]: do you need the `\` before the `.`?

Comment (krzyzacy) [reply kubetest/extract.go]: isn't `.` by itself will match any single character?

Comment (ixdy) [reply kubetest/extract.go]: I think because this is a character class `.` doesn't have special meaning here - it's just a period, nothing else.

Comment (fejta) [reply kubetest/extract.go]: Jeff can you suggest a unit test that demonstrates the problem?

Comment (ixdy) [reply kubetest/extract.go]: I did some one-off tests and I don't think having the `\` before the `.` has any effect, either negative or positive. I think @krzyzacy already added a unit test case which covers this.

Comment (krzyzacy) [reply kubetest/extract.go]: oh whoops, I missed the important `the`, yeah that probably don't take any effect, but `-` does

File: kubetest/extract.go [hunk 4]
```diff
@@ -68,16 +68,16 @@ func (l *extractStrategies) String() string {
...
-		`^(gs://.*)$`:               gcs,
-		`^(bazel/.*)$`:              bazel,
+		`^(local)`:                    local,
+		`^gke-?(staging|test)?$`:      gke,
+		`^gci/([\w-]+)$`:              gci,
+		`^gci/([\w-]+)/(.+)$`:         gciCi,
+		`^ci/(.+)$`:                   ci,
+		`^release/(latest.*)$`:        rc,
+		`^release/(stable.*)$`:        stable,
+		`^(v\d+\.\d+\.\d+[\w.\-+]*)$`: version,
```
Comment (ixdy) [inline kubetest/extract.go]: (you don't need to change this, but a tip: if you put `-` at the start of the character class, you don't need to escape it, i.e.
 something like `[-\w.+]` would work.)

Comment (krzyzacy) [reply kubetest/extract.go]: I see, good to know, thanks!

**Comment by krzyzacy** [comment]
/hold cancel
probably updating image together with future changes

**Pull request merged**